blockchain/chaingen/generator.go

```diff
@@ -2085,9 +2116,10 @@ func (g *Generator) NextBlock(blockName string, spend *SpendableOut, ticketSpend
 		// it.
 		g.originalParents[blockHash] = prevHash
 	}
-	g.connectLiveTickets(&blockHash, nextHeight, ticketWinners,
+	g.connectLiveTickets(&blockHash, block.Header.Height, ticketWinners,
```

Comment:
I'm fairly certain this shouldn't be changed.  Just because the block header height changed making it invalid does not mean the rest of the state should be made invalid to match.